src/AI-DataFrameInspector/AISpDataFramePresenter.class.st

```diff
@@ -0,0 +1,250 @@
+	setRowNames ifNil: [ setRowNames := false ].
+	super initialize
+]
+
+{ #category : #initialization }
+AISpDataFramePresenter >> initializeDataFramePresenter [
+
+	dataFramePresenter := self newTable.
+
+	dataFramePresenter addColumn: (SpIndexTableColumn new
+			 title: '#';
+			 sortFunction: #yourself ascending;
+			 beNotExpandable;
+			 yourself).
+	dataFrame rowNames
+	= (1 to: dataFrame numberOfRows) asOrderedCollection ifFalse: [
+		setRowNames := true.
+		dataFramePresenter addColumn: (SpStringTableColumn
+				 title: ''
+				 evaluated: [ :rowWithName | rowWithName at: 1 ]) ].
+
+	dataFrame columnNames doWithIndex: [ :headerName :columnIndex |
+		dataFramePresenter addColumn: (SpStringTableColumn
+				 title: headerName
+				 evaluated: [ :rowWithName | rowWithName at: columnIndex + 1 ]) ].
+	dataFramePresenter beResizable.
+
+	dataFramePresenter items: dataFrame asArrayOfRowsWithName
+]
+
+{ #category : #initialization }
+AISpDataFramePresenter >> initializeNewDataFramePresenter [
+
+	newdataFramePresenter := self newTable.
+
+	newdataFramePresenter addColumn: (SpIndexTableColumn new
+			 title: '#';
+			 sortFunction: #yourself ascending;
+			 beNotExpandable;
+			 yourself).
+	setRowNames ifTrue: [
+		newdataFramePresenter addColumn: (SpStringTableColumn
+				 title: ''
+				 evaluated: [ :rowWithName | rowWithName at: 1 ]) ].
+
+	dataFrame columnNames doWithIndex: [ :headerName :columnIndex |
+		newdataFramePresenter addColumn: (SpStringTableColumn
+				 title: headerName
+				 evaluated: [ :rowWithName | rowWithName at: columnIndex + 1 ]) ].
+	newdataFramePresenter beResizable.
+
+	newdataFramePresenter items: dataFrame asArrayOfRowsWithName
+]
+
+{ #category : #initialization }
+AISpDataFramePresenter >> initializePresenters [
+
+	self initializeDataFramePresenter.
+	self initializeToolbarPresenter.
+	self initializeSortPresenter
+]
+
+{ #category : #initialization }
+AISpDataFramePresenter >> initializeSortPresenter [
+
+	windowPresenter := SpWindowPresenter new.
+	sortPresenter := self
+		                 instantiate: AISpSortTablePresenter
+		                 on: dataFrame
+]
+
+{ #category : #initialization }
+AISpDataFramePresenter >> initializeToolbarPresenter [
+
+	editButton := self newButton
+		              label: 'Edit';
+		              icon: (self iconNamed: #editIcon);
+		              yourself.
+	searchBar := self newTextInput placeholder: '          search'.
+	sortButton := self newButton
+		              label: 'sort';
+		              icon: (self iconNamed: #smallOpen);
+		              yourself
+]
+
+{ #category : #searching }
+AISpDataFramePresenter >> searchBarAction [
+
+	searchBar text ifEmpty: [
+		self initializeDataFramePresenter.
+		self layout replace: newdataFramePresenter with: dataFramePresenter ].
+	searchBar text ifNotEmpty: [
+		newdataFramePresenter := dataFramePresenter.
+		self filterTableWithSearchQuery: searchBar text.
+		self layout replace: dataFramePresenter with: newdataFramePresenter ]
```

Comment:
```suggestion
	searchBar text
		ifEmpty: [
			self initializeDataFramePresenter.
			self layout replace: newdataFramePresenter with: dataFramePresenter ].
		ifNotEmpty: [ :text |
			newdataFramePresenter := dataFramePresenter.
			self filterTableWithSearchQuery: text.
			self layout replace: dataFramePresenter with: newdataFramePresenter ]
```

Readability

Reply:
Yes it is more readable this way.
I guess you added a period after the ifEmpty[ ] block by mistake, I took care of this in my commit.